nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/migrator/AbstractRepositoryMigrator.java

```diff
@@ -58,21 +64,41 @@ protected void reportProgress(String repositoryName, String message, long num, l
         reportProgress(String.format("[%s] %s", repositoryName, message), num, total);
     }
 
+    /**
+     * @since 11.1
+     */
+    protected void migrateSession(String step, MigrationContext context, CoreSession session) {
```

Comment:
Could you move it after the second `migrateRepository`, as the latest calls `migrateSession`.